core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2010,11 +2010,21 @@ public void deleteInternal(RpcContext rpcContext, LockedInodePath inodePath,
       }
 
       if (!failedUris.isEmpty()) {
-        Collection<String> messages = failedUris.stream()
-            .map(pair -> String.format("%s (%s)", pair.getFirst(), pair.getSecond()))
```

Comment:
The map-collect will generate too many small strings. Also on the collection phase, the ArrayList will have auto-sizing and many array copies. The collected list will be joined into one string at last, so the streaming is essentially just a syntactic sugar resource waste.